### DISCUSSION TITLE
Warnings about impure methods

### DISCUSSION THREAD

**Discussion by sdether** [BeanCheeseBurrito/Flecs.NET#79] [Q&A] [answered]
I'm storing my `World` instance as a readonly field in some classes and when I access the world, _JetBrains Rider_ annotates the calls with `Possibly impure struct method called on readonly variable: struct value always copied before invocation`.

I'm assuming all the methods are _pure_ in the sense that they don't actually change the `World` struct. As far as I can tell `World` only has a single member and its' a pointer to the native world, so any _mutation_ is happening in the native Flecs code. But I wanted to make sure.

**Comment by BeanCheeseBurrito** [comment, marked as answer]
None of the world methods modify the ``World`` struct, except for ``World.Dispose()`` and ``World.Reset()``. In the next release I'll change the ``World`` struct to be readonly and get rid of any code that modifies the world handle.
